code/src/App.js

```diff
@@ -1,9 +1,72 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+
+import ThoughtForm from './components/ThoughtForm.js'
+import ThoughtList from './components/ThoughtList.js'
+
+import { API_URL_THOUGHTS } from './reusable/urls'
+import { API_URL_HEARTS } from './reusable/urls'
 
 export const App = () => {
+  const [thoughtList, setThoughtList] = useState([]) // state properties initialized as an empty array
+  const [thoughtNew, setThoughtNew] = useState('') // state properties initialized as an empty string
+
+  useEffect(() => {    // useEffect hook
+    fetchThoughtList()
+  }, [])
+
+  const fetchThoughtList = () => {  // GET fetch request for resent thoughts
+    fetch(API_URL_THOUGHTS)
+      .then(res => res.json()) // response unpacked
+      .then(thoughts => setThoughtList(thoughts)) // data recieved and applied
+      .catch(err => console.error(err))
+  }
+
+  const handleThoughtNewChanged = (event) => { // handle indicates that this handler will be passed to a component and used there.
+    setThoughtNew(event.target.value)
+  }
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault()
+    window.location.reload()
+    event.target.reset()
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: thoughtNew })
+    };
+
+    fetch(API_URL_THOUGHTS, options)
+      .then(res => res.json())
+      //.then(receivedThought => setThoughtList([...thoughtList, receivedThought]))
```

Comment:
I think the code you commented out is the right one to update the already existing list of thoughts with a new one. Now it seems that this functionality is not working :(